utility/icon.go

```diff
@@ -0,0 +1,33 @@
+package utility
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"path/filepath"
+	"sort"
+
+	"github.com/bitrise-io/bitrise-init/models"
+	"github.com/bitrise-io/go-utils/sliceutil"
+)
+
+// ConvertPathsToUniqueFileNames returns a sorted array of unique icons.
+func ConvertPathsToUniqueFileNames(absoluteIconPaths []string, basepath string) (models.Icons, error) {
```

Comment:
NOTE: this function does a bit more then i expected based on the function name, what about: `CreateIconDescriptors?`